boilerplate/_lib/ensure.sh

```diff
@@ -3,17 +3,27 @@ set -euo pipefail
 
 GOLANGCI_LINT_VERSION="1.30.0"
 DEPENDENCY=${1:-}
+GOOS=$(go env GOOS)
 
 case "${DEPENDENCY}" in
-    golangci-lint)
-        GOPATH=$(go env GOPATH)
-        if [ ! -f "${GOPATH}/bin/golangci-lint" ]; then
-            DOWNLOAD_URL="https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64.tar.gz"
-            curl -sfL "${DOWNLOAD_URL}" | tar -C "${GOPATH}/bin" -zx --strip-components=1 "golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64/golangci-lint"
+golangci-lint)
+    GOPATH=$(go env GOPATH)
+    if which golangci-lint ; then
+        exit
```

Comment:
This doesn't ensure we're using `$GOLANGCI_LINT_VERSION`. As written, there are several ways that could go awry even when we're running the `gocheck` target in a container -- e.g. if the base container image we're using already has the binary (at the wrong version).

So IMO we really ought to include a version check here. Something like:
```suggestion
    if which golangci-lint && [[ $(golangci-lint --version) == *" v$GOLANGCI_LINT_VERSION "* ]]; then
        exit
```